client/web/admin/src/views/System/Connection/Editor.vue

```diff
@@ -239,6 +252,14 @@ export default {
           this.processing = false
         })
     },
+
+    checkUnsavedChanges (next) {
+      if (!this.$route.path.includes('/new')) {
```

Comment:
Im thinkin it might make sense to issue the warning always if on new, since you always have unsaved changes there. That means you dont need to check if config changed, you just always warn.